### PR TITLE
ref(kube): use external api types where possible

### PR DESCRIPTION
**Pull request by adamreese**

### CODE REVIEW

File: pkg/kube/client.go [hunk 1]
```diff
@@ -163,7 +162,7 @@ func (c *Client) Get(namespace string, reader io.Reader) (string, error) {
 		return "", err
 	}
 
-	var objPods = make(map[string][]core.Pod)
+	var objPods = make(map[string][]v1.Pod)
```
Comment (mortent) [inline pkg/kube/client.go]: This breaks the printing of pod status. The printer only outputs resource-specific information when the internal type is used. With this change, only the age will be printed for pods.

Ideally the internal types shouldn't be used in the helm codebase at all, but for now it is the only way to get the proper output from the printer. There is an ongoing effort to fix this in the kubernetes codebase, but it probably won't be coming until 1.14.

File: pkg/kube/client.go [hunk 2]
```diff
@@ -714,8 +713,8 @@ func (c *Client) getSelectRelationPod(info *resource.Info, objPods map[string][]
 	}
 
 	for _, externalPod := range pods.Items {
-		pod := core.Pod{}
-		legacyscheme.Scheme.Convert(&externalPod, &pod, nil)
+		pod := v1.Pod{}
+		scheme.Scheme.Convert(&externalPod, &pod, nil)
```
Comment (mortent) [inline pkg/kube/client.go]: I don't think there is any reason to do any conversion here. `externalPod` is already of type v1.Pod.

Comment (adamreese) [reply pkg/kube/client.go]: good catch, fixed

File: pkg/kube/client.go (hunk 1 above)
Comment (adamreese) [reply pkg/kube/client.go]: objects are converted to internal api before printing

**Review by technosophos — APPROVED** [review]
LGTM

**Pull request merged**